scaaml/capture/input_generators/attack_point_iterator_models.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Pydantic models for the attack point iterator."""
+from pydantic import BaseModel, model_validator
+from typing import Any, Dict, Iterator, Literal, List, TypeAlias, Union
+
+from scaaml.capture.input_generators.attack_point_iterator_exceptions import ListNotPrescribedLengthException
+from scaaml.capture.input_generators.input_generators import balanced_generator, unrestricted_generator
+
+AttackPointIteratorT = Iterator[Dict[str, Any]]
+
+
+class ConstantIteratorModel(BaseModel):
+    """
+    Attack point iterator pydantic model that iterates over a constant.
+    
+    Args:
+        operation (str): The operation of the iterator
+        represents what the iterator does and what 
+        arguments should be present. This is only used once to
+        double check if the operation is the correct one.
+
+        name (str): The name represents the key name of the value.
+
+        length (int): The prescribed length for each list in values.
+        If one of the lists isn't the same length as this variable
+        it will raise an ListNotPrescribedLengthException.
+
+        values (List[List[int]]): List of lists of ints that gets
+        iterated through.
+    """
+    operation: Literal['constants']
```

Comment:
Move down and put a default value.
```suggestion
    operation: Literal["constants"] = "constants"
```